src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -454,11 +454,24 @@ private boolean hasFallThroughComment(DetailAST currentCase) {
      * @return true if relief comment found
      */
     private boolean hasReliefComment(DetailAST ast) {
-        return Optional.ofNullable(getNextNonCommentAst(ast))
-                .map(DetailAST::getPreviousSibling)
-                .map(previous -> previous.getFirstChild().getText())
-                .map(text -> reliefPattern.matcher(text).find())
-                .orElse(Boolean.FALSE);
```

Comment:
More just curious why we can't keep this original structure of the code and have to revert to basically a pre-Java 8 version of the code?

Reply:
We could certainly create a list of siblings here and filter them to make this method more concise and avoid mutable data.

Reply:
Sure. I am not very familiar with this new style as well. I will try to change this to the new style.

Reply:
Done, please have a look.

```java
result = Stream.iterate(nonCommentAst.getPreviousSibling(),
                            Objects::nonNull, DetailAST::getPreviousSibling)
                    .takeWhile(sibling -> sibling.getLineNo() == prevLineNumber)
                    .map(DetailAST::getFirstChild)
                    .filter(Objects::nonNull)
                    .anyMatch(firstChild -> reliefPattern.matcher(firstChild.getText()).find());
```

I didn't really create a list of siblings, but I used `iterate` to iterate over all thes sblings. I think this way might be more efficient.

Reply:
@nrmancuso Please assist.

@Lmh-java If you think this a 1 to 1, then regression should return the same results.

Reply:
@Lmh-java very good, takeWhile is a great stream operation to use here. 🚀

Reply:
>very good, takeWhile is a great stream operation to use here. 🚀

Thank you! 

>If you think this a 1 to 1, then regression should return the same results.

https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/7e9dba3_2024170824/reports/diff/index.html
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8a0a354_2024060435/reports/diff/index.html

@rnveach These two regression report has the same result. Please take a look.